### PR TITLE
feat(eslint-plugin): add extension rule `no-duplicate-imports`

### LINKED ISSUE #2315: [no-duplicate-imports]  false positive rule on "import type"
**Repro**

```ts
import type { Foo } from './foo'
import { foo } from './foo'
```

reports `eslint(no-duplicated-imports)` error

### PR DESCRIPTION
**Pull request by yeonjuan**
fixes #2315

I made an extension rule which distinguishes type-only import from standard import. :)

### CODE REVIEW

**Review by bradzacher — APPROVED** [review]
LGTM - thanks for your contribution!

File: packages/eslint-plugin/src/rules/no-duplicate-imports.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      node: TSESTree.ExportNamedDeclaration | TSESTree.ExportAllDeclaration,
+    ): void {
+      if (isStringLiteral(node.source)) {
+        const value = node.source.value;
+        if (typeExports.has(value)) {
+          report('exportType', node, value);
+        }
+        if (typeImports.has(value)) {
+          report('exportTypeAs', node, value);
+        }
```
[the remark this answers is not in the document]
Comment (yeonjuan) [reply packages/eslint-plugin/src/rules/no-duplicate-imports.ts:88]: @bradzacher right! This rule seems just focus on duplicate module sources.🤣

**Comment by richard-viney** [comment]
I've run into some false positives with this rule when used with `import type`.

Specifically, the following code triggers this new rule:

```ts
import type Foo from "my-module"
import type { Bar } from "my-module"
```

But it is not actually possible to write

```ts
import type Foo, { Bar } from "my-module"
```

because that results in the TypeScript error

```
A type-only import can specify a default import or named bindings, but not both.ts(1363)
```

This rule may need some additional logic to handle this case.

**Comment by bradzacher** [comment]
please file a new issue.

**Comment by richard-viney** [comment]
New issue is here: https://github.com/typescript-eslint/typescript-eslint/issues/2636

File: packages/eslint-plugin/tests/rules/no-duplicate-imports.test.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import rule from '../../src/rules/no-duplicate-imports';
+import { RuleTester } from '../RuleTester';
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('no-dupe-class-members', rule, {
```
Comment (ginger-kang) [inline packages/eslint-plugin/tests/rules/no-duplicate-imports.test.ts:8]: Hi :)
Isn't this rule id should be `no-duplicate-imports`, not `no-dupe-class-members`?

**Pull request merged**